### PR TITLE
Add support for multimedia keys

### LINKED ISSUE #12986: Keys on physical keyboard on android are mixed up
## Description

Backspace key is the same as delete key on android with a physical keyboard.

### Expected behavior

Delete key and backspace act separate.
 
### Actual behavior

Backspace and delete act the same

### Steps to reproduce the bug

1. Load a dosbox core on android command prompt.
2. Type something with a physical keyboard then go to the beginning of the comand.
3. Press backspace.

- RetroArch: 1.9.9 stable

### Environment information

- OS: [android version 10]

### LINKED ISSUE #14298: Keyboard volume up, down, mute keys not detected
# First and foremost consider this:
- Only RetroArch bugs should be filed here. Not core bugs or game bugs
- This is not a forum or a help section, this is strictly developer oriented

## Description

Settings -> Input -> Hotkeys does not recognize physical volume up, down and mute keys. The retroarch.cfg file also has no support for using these keys.
Physical volume  up, down and mute keys can be found on some keyboards and also on remotes which present to the OS as a keyboard. I'm not talking about the plus and minus keys.

### Expected behavior

In the Settings  -> Input -> Hotkeys menu if you press the volume up, down and mute keys these should be detected
In the retroarch.cfg file the volume up, down and mute keys could be added to the list of special keys starting at line 449

### Actual behavior

In the Settings  -> Input -> Hotkeys menu if you press the volume up, down or mute keys nothing happens and the interface waits for a key press.
In the retroarch.cfg file there is no way to assign the volume up, down and mute keys to an input action.

### Steps to reproduce the bug

Open the Settings -> Input -> Hotkeys menu
Choose Volume Up (doesn't matter which action you choose here)
The interface shows a windows with 'press keyboard, mouse or joystick' with a timeout
Press the physical volume up, down or mute key on a keyboard
The key press is not detected

### Bisect Results

[Try to bisect and tell us when this started happening]

### Version/Commit
You can find this information under Information/System Information

- RetroArch: 9b282aa

### Environment information

- OS: Lakka (RPi3.aarch64-4.2)
- Compiler: [In case you are running local builds]

### PR DESCRIPTION
**Pull request by zoltanvb**
## Description

Add a list of extra keys commonly found on "multimedia" keyboards: new RETROK_ entries and mappings for SDL, X11, Wayland, dinput, winraw drivers. Extend keyboard tester to cover also these keys.

Due to wide HW variety, there is no definite list, I used [this collection](https://www.autohotkey.com/docs/v1/KeyList.htm#multimedia) which also agrees with my own memories of such a keyboard from several years ago.

Since the OS usually catches these keys before they reach RetroArch, the usefulness is mostly for Lakka or for Linux+KMS mode.
Using a programmable keyboard, I was able to hit most of the keys, see the last row of the extended keyboard tester:
RetroPad Remote-240505-192035

The missing ones were not present in the mapping software. Assigning them to a hotkey was also tried.

Outline for adding a new key was used from the [OEM_102 commit](https://github.com/libretro/RetroArch/commit/b592f254b9de165863c9a9e8c914a36536fbc9fe).

Leaving it in draft for a few days, as feedback would be welcome on:
- should some new keys be added? sleep/power, F16..F24?
- any reason to add such keys to console / phone platforms?
- any reason to add such keys to Apple platforms?
- is app1 / app2 key being more used on actual physical keyboards, or rather "my computer"/"calculator"? There are several platforms where both variants are present.

## Related Issues

Fixes #14298 
Fixes #12986 (not related to new keys, but it is in the same mapping file)

## Related Pull Requests

#16390

### CODE REVIEW

**Comment by zoltanvb** [comment]
No strong opinions, so from my point of view it is OK as it is.

File: cores/libretro-net-retropad/net_retropad_core.c [hunk 1]
```diff
@@ -586,7 +586,7 @@ static void open_UDP_socket()
    socket_target_t in_target;
 
    if (s && s != SOCKET_ERROR)
-      close(s);
+      socket_close(s);
```
Comment (RobLoach) [inline cores/libretro-net-retropad/net_retropad_core.c:589]: This is the only change that looks unrelated, but correct?

**Review by RobLoach — APPROVED** [review]
Thanks for the retropad updates :+1:  I was able to run the core. My multimedia keys were caught by the OS, but things were looking good.

**Comment by LibretroAdmin** [comment]
Sorry for the long wait @zoltanvb .

Looks good.

File: cores/libretro-net-retropad/net_retropad_core.c (hunk 1 above)
Comment (zoltanvb) [reply cores/libretro-net-retropad/net_retropad_core.c:589]: Same as in https://github.com/libretro/RetroArch/pull/16494/files, forgot to rebase, sorry. But this is already in via #16494 .

**Pull request merged**